javascripts/discourse/components/show-raw-button.gjs

```diff
@@ -0,0 +1,40 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import FullscreenCode from "discourse/components/modal/fullscreen-code";
+import { ajax } from "discourse/lib/ajax";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default class ShowRawButton extends Component {
+  static hidden = true;
+
+  @service modal;
+
+  @action
+  async showRaw() {
+    try {
+      const response = await ajax(`/posts/${this.args.post.id}/raw`, {
+        dataType: "text",
+      });
+      this.modal.show(FullscreenCode, {
```

Comment:
I know we don't do it often, but I think modal.show returns a promise, so might be nice to use it